### ISSUE TITLE
new 'Pick' type throwing errors everywhere

### ISSUE THREAD

**Issue by jpike88** [tgriesser/knex#3274]
knex@0.18.0-next1

I get that typing is nice, but I now have 181 errors in my project due to errors like this:

Error:TS2339 Property 'id' does not exist on type 'Pick<any, "assessment.*">'.

For lines like this:

```const assessmentAttrs = await sql.knex('attribute_assessment_map').whereIn('assessmentID', assessments.map(a => a.id));```

This can't be working correctly.

**Comment by lorefnon** [comment]
Can you please also post how are you retrieving the assesments array ?

**Comment by jpike88** [comment]
```
const assessments = await sql.knex('assessment')
		.join('assessment_clientGroup_map', 'assessment_clientGroup_map.assessmentID', 'assessment.id')
		.whereIn('assessment_clientGroup_map.clientGroupID', [1, 2, 3])
		.select('assessment.*');
	const assessmentAttrs = await sql.knex('attribute_assessment_map')
		.whereIn('assessmentID', assessments.map(a => a.id));
```

**Comment by lorefnon** [comment]
This version does not seem to exist, has it been unpublished ? 
I can not reproduce the error in either master or the latest 0.17.x release.

**Comment by kibertoad** [comment]
@lorefnon it wasn't published, just use master. Since no babel is involved, knex git branches can finally be used in npm.

**Comment by kibertoad** [comment]
@jpike88 can you still reproduce it with latest master?

**Comment by jpike88** [comment]
my package lock is on ```github:tgriesser/knex#4b6429ebf958b6a3211cb2e2c85ae2d9338b3867```, so yes the problem is on latest commit

**Comment by kibertoad** [comment]
Released in 0.18.0-next1
